### PR TITLE
Bugfix/wait zero doesnt sleep

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
Fix when using bWorkerFlushAfterOutgoingNetworkOp and an OpsUpdateRate > 1000. This causes the event to not sleep and the worker thread is a busy loop. 

This hasn't gone out in a release yet so not adding a release note. 

#### Primary reviewers
@m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h [hunk 1]
```diff
@@ -33,7 +33,8 @@ class WorkerConnectionCoordinator
 		if (Event.IsValid())
 		{
 			FTimespan WaitTime = FTimespan::FromSeconds(WaitSeconds);
-			Event->Wait(WaitTime);
+			int32_t TimeToWaitMS = FGenericPlatformMath::Max(1, static_cast<int32>(WaitTime.GetTotalMilliseconds()));
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h:36]:
```suggestion
			int32 TimeToWaitMS = FGenericPlatformMath::Max(1, static_cast<int32>(WaitTime.GetTotalMilliseconds()));
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 2]
```diff
@@ -21,7 +21,14 @@ void USpatialWorkerConnection::SetConnection(Worker_Connection* WorkerConnection
 		if (OpsProcessingThread == nullptr)
 		{
 			bool bCanWake = SpatialGDKSettings->bWorkerFlushAfterOutgoingNetworkOp;
-			ThreadWaitCondition.Emplace(bCanWake, 1.0f / GetDefault<USpatialGDKSettings>()->OpsUpdateRate);
+			float WaitTimeS = 1.0f / (GetDefault<USpatialGDKSettings>()->OpsUpdateRate);
+			int32 WaitTimeMs = static_cast<int32>(FTimespan::FromSeconds(WaitTimeS).GetTotalMilliseconds());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:25]: Is it actually possible for (1/x) to be == 0 ?

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:25]: Nah but its the integer conversion which breaks it - the sleep functions take milliseconds in integer form

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:25]: Ahh gotcha

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 3]
```diff
@@ -21,7 +21,14 @@ void USpatialWorkerConnection::SetConnection(Worker_Connection* WorkerConnection
...
+			if (WaitTimeMs <= 0)
+			{
+				UE_LOG(LogSpatialWorkerConnection, Warning, TEXT("Clamping wait time for worker ops thread, minimum frequency is 1 millisecond."));
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:28]:
```suggestion
				UE_LOG(LogSpatialWorkerConnection, Warning, TEXT("Clamping wait time for worker ops thread, minimum rate is 1 millisecond."));
```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h [hunk 4]
```diff
@@ -19,25 +19,25 @@ struct FEventDeleter
 class WorkerConnectionCoordinator
 {
 	TUniquePtr<FEvent, FEventDeleter>		Event;
-	float									WaitSeconds;
+	int32									WaitTime;
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h:22]:
```suggestion
	int32 WaitTime;
```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h [hunk 5]
```diff
@@ -19,25 +19,25 @@ struct FEventDeleter
 class WorkerConnectionCoordinator
 {
 	TUniquePtr<FEvent, FEventDeleter>		Event;
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h:21]:
```suggestion
	TUniquePtr<FEvent, FEventDeleter> Event;
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 6]
```diff
@@ -21,7 +21,14 @@ void USpatialWorkerConnection::SetConnection(Worker_Connection* WorkerConnection
 		if (OpsProcessingThread == nullptr)
 		{
 			bool bCanWake = SpatialGDKSettings->bWorkerFlushAfterOutgoingNetworkOp;
-			ThreadWaitCondition.Emplace(bCanWake, 1.0f / GetDefault<USpatialGDKSettings>()->OpsUpdateRate);
+			float WaitTimeS = 1.0f / (GetDefault<USpatialGDKSettings>()->OpsUpdateRate);
+			int32 WaitTimeMs = static_cast<int32>(FTimespan::FromSeconds(WaitTimeS).GetTotalMilliseconds());
+			if (WaitTimeMs <= 0)
+			{
+				UE_LOG(LogSpatialWorkerConnection, Warning, TEXT("Clamping wait time for worker ops thread to the minimum rate to 1ms."));
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:28]:
```suggestion
				UE_LOG(LogSpatialWorkerConnection, Warning, TEXT("Clamping wait time for worker ops thread to the minimum rate of 1ms."));
```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h [hunk 7]
```diff
@@ -18,26 +18,26 @@ struct FEventDeleter
 */
 class WorkerConnectionCoordinator
 {
-	TUniquePtr<FEvent, FEventDeleter>		Event;
-	float									WaitSeconds;
+	TUniquePtr<FEvent, FEventDeleter> Event;
+	int32 WaitTimeMS;
 public:
-	WorkerConnectionCoordinator(bool bCanWake, float InWaitSeconds)
+	WorkerConnectionCoordinator(bool bCanWake, int32 InWaitMs)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h:24]:
```suggestion
	WorkerConnectionCoordinator(bool bCanWake, int32 InWaitMS)
```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h [hunk 8]
```diff
@@ -18,26 +18,26 @@ struct FEventDeleter
...
 		: Event(bCanWake ? FGenericPlatformProcess::GetSynchEventFromPool() : nullptr)
-		, WaitSeconds(InWaitSeconds)
+		, WaitTimeMS(InWaitMs)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/WorkerConnectionCoordinator.h:26]:
```suggestion
		, WaitTimeMS(InWaitMS)
```

**Pull request merged**